pkg/util/containers/providers/cgroup/cgroup_metrics.go

```diff
@@ -266,36 +266,58 @@ func (c ContainerCgroup) CPUPeriods() (throttledNr uint64, throttledTime float64
 
 // CPULimit would show CPU limit for this cgroup.
 // It does so by checking the cpu period and cpu quota config
-// if a user does this:
+// or cpuset if CPUs are pinned.
+// If a user does this:
 //
 //	docker run --cpus='0.5' ubuntu:latest
 //
 // we should return 50% for that container.
 //
 // However cfs_period_us is per CPU, which means that
 //
-// docker run --cpu='2' ubuntu:latest
+// docker run --cpus='2' ubuntu:latest
 //
 // Will yield 200% (cfs_period_us = 100000, cfs_quota_us = 200000)
 //
+// If a user does:
+//
+// docker run --cpuset-cpus='1,3' ubuntu:latest
+//
+// we should return 200%
+//
+// In the case that both CFS quota and CPU sets are defined, we take the minimum.
+//
 // If the limits files aren't available (on older version) then
 // we'll return the default value of numCPU * 100.
 func (c ContainerCgroup) CPULimit() (float64, error) {
-	limit := float64(system.HostCPUCount()) * 100.0
+	defaultLimit := float64(system.HostCPUCount()) * 100.0
+	limitFromCPUSet := float64(-1)
+	limitFromQuota := float64(-1)
+
+	cpusetFile := c.cgroupFilePath("cpuset", "cpuset.cpus")
+	cpuLines, err := readLines(cpusetFile)
+	if os.IsNotExist(err) {
+		log.Debugf("Missing cgroup file: %s", cpusetFile)
+	} else {
```

Comment:
shouldn't we handle the case where `err != nil` and `os.IsNotExist(err)` is false ?